frontend/vite-project/src/components/card-component/Sample.tsx

```diff
@@ -0,0 +1,150 @@
+import { useState } from 'react';
+import ProductCard from "../card-component/ProductCard";
+import ProductCardCartMobile from './ProductCardCartMobile';
+import ProductCardCart from './ProductCardCart';
+
+const Sample = () => {
+    const productCatalog = [
+        {
+          id: 1,
+          imageUrl: "https://images.pexels.com/photos/54082/carrots-vegetables-food-orange-54082.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1",
+          seedName: "Carrot",
+          price: 10,
+        },
+        {
+          id: 2,
+          imageUrl: "https://images.pexels.com/photos/1327838/pexels-photo-1327838.jpeg",
+          seedName: "Tomato",
+          price: 12,
+        },
+        {
+          id: 3,
+          imageUrl: "https://images.pexels.com/photos/128420/pexels-photo-128420.jpeg",
+          seedName: "Zucchini",
+          price: 8,
+        },
+    ];
+      
+
+  const [cart, setCart] = useState<{ id: number; quantity: number }[]>([]);
+
+  const handleAddToCart = (id: number, quantity: number) => {
+    if (quantity === 0) return;
+
+    setCart(prev => {
+      const existing = prev.find(item => item.id === id);
+      if (existing) {
+        return prev.map(item =>
+          item.id === id ? { ...item, quantity: item.quantity + quantity } : item
+        );
+      }
+      return [...prev, { id, quantity }];
+    });
+  };
+
+  const handleRemove = (id: number) => {
+    setCart(prev => prev.filter(item => item.id !== id));
+  };
+
+  const handleQuantityChange = (id: number, newQuantity: number) => {
+    setCart(prev =>
+      prev.map(item =>
+        item.id === id ? { ...item, quantity: newQuantity } : item
+      )
+    );
+  };
```

Comment:
```suggestion
  const handleQuantityChange = (id: number, newQuantity: number) => {
   if(newQuantity === 0 ){
    handleRemove(id)
   }
     setCart(prev =>
      prev.map(item =>
        item.id === id ? { ...item, quantity: newQuantity } : item
      )
    );
  };
```
Could also consider limiting the cart quantity so it doesn't go below one, but using the handleRemove function here works quite well and keeps the logic clean.